crates/wasmtime/src/memory.rs

```diff
@@ -252,12 +252,19 @@ impl Memory {
             store.0.async_support(),
             "cannot use `new_async` without enabling async support on the config"
         );
-        store.on_fiber(|store| Memory::_new(store.0, ty)).await?
+        store
+            .on_fiber(|store| Self::_new(store.0, ty, None))
+            .await?
     }
 
-    fn _new(store: &mut StoreOpaque, ty: MemoryType) -> Result<Memory> {
+    /// Helper function for attaching the memory to a "frankenstein" instance
+    fn _new(
+        store: &mut StoreOpaque,
+        ty: MemoryType,
+        preallocation: Option<wasmtime_runtime::SharedMemory>,
```

Comment:
The `vmimport` constructor doesn't need to create a `Memory` through which inserts into the memories table within the `Store`, so I think that the call to the trampoline helper here can be inlined into `vmimport` and then the instance handle can be used to extract the memory directly rather than going through `Memory`